src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -463,10 +466,36 @@ export class PageControllerBase {
 
       viewModel.backLink = this.getBackLink(progress)
 
+      const missingEmailWarning = await this.buildMissingEmailWarningModel(
+        request,
+        isStartPage
+      )
+
+      if (missingEmailWarning) {
+        return h.view(this.viewName, { ...viewModel, ...missingEmailWarning })
+      }
+
       return h.view(this.viewName, viewModel)
     }
   }
 
+  async buildMissingEmailWarningModel(request: Request, isStartPage?: boolean) {
+    const { params } = request
+    // Warn the user if the form has no notification email set only on start page and summary page
+    if (isStartPage || params.path === 'summary') {
+      const { slug } = params as { slug: string }
+      const { notificationEmail } = await getFormMetadata(slug)
```

Comment:
Glad to see we're not wasting API calls on every page request and only doing it when required